x-pack/plugins/case/server/connectors/case/index.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { curry } from 'lodash';
+
+import { KibanaRequest } from 'kibana/server';
+import { ActionTypeExecutorResult } from '../../../../actions/common';
+import { CasePatchRequest, CasePostRequest } from '../../../common/api';
+import { createCaseClient } from '../../client';
+import { CaseExecutorParamsSchema, CaseConfigurationSchema } from './schema';
+import {
+  CaseExecutorParams,
+  CaseExecutorResponse,
+  ExecutorSubActionCreateParams,
+  ExecutorSubActionUpdateParams,
+  ExecutorSubActionAddCommentParams,
+  CaseActionType,
+  CaseActionTypeExecutorOptions,
+} from './types';
+
+import { GetActionTypeParams } from '..';
+
+const supportedSubActions: string[] = ['create', 'update', 'addComment'];
+
+// action type definition
+export function getActionType({
+  logger,
+  caseService,
+  caseConfigureService,
+  userActionService,
+}: GetActionTypeParams): CaseActionType {
+  return {
+    id: '.case',
+    minimumLicenseRequired: 'gold',
+    name: 'Case',
+    validate: {
+      config: CaseConfigurationSchema,
+      params: CaseExecutorParamsSchema,
+    },
+    executor: curry(executor)({ logger, caseService, caseConfigureService, userActionService }),
+  };
+}
+
+// action executor
+
+async function executor(
+  { logger, caseService, caseConfigureService, userActionService }: GetActionTypeParams,
+  execOptions: CaseActionTypeExecutorOptions
+): Promise<ActionTypeExecutorResult<CaseExecutorResponse | {}>> {
+  const { actionId, params, services } = execOptions;
+  const { subAction, subActionParams } = params as CaseExecutorParams;
+  let data: CaseExecutorResponse | null = null;
+
+  const { savedObjectsClient } = services;
+  const caseClient = createCaseClient({
+    savedObjectsClient,
+    request: {} as KibanaRequest,
```

Comment:
I'm confused about why we're passing an empty object and typing it. Instead can this value be undefined? Can we change the type?